.changeset/thin-seas-try.md

```diff
@@ -0,0 +1,12 @@
+---
+"@comet/cms-admin": patch
+"@comet/admin": patch
+"@comet/cms-api": patch
+---
+
+Handle unauthorized and unauthenticated correctly in error dialog
+
+The error dialog now presents screens according to the current state. Required to work in all conditions:
+
+-   `ErrorDialogHandler` must be beneath `MuiThemeProvider` and `IntlProvider`
+-   `CurrentUserProvider` must be beneath or parallel to `ErrorDialogHandler`
```

Comment:
Would it work without making this change? Otherwise this is a breaking change

Reply:
<img width="1683" alt="image" src="https://github.com/vivid-planet/comet/assets/48853629/f142aa9f-3a3e-41ed-9ef4-9878926bf5e9">

I just tried this locally and it fails with an unrecoverable error

Reply:
> `ErrorDialogHandler` must be beneath `MuiThemeProvider` and `IntlProvider`

This isn't an actual case. The `ErrorDialogHandler` was beneath `MuiThemeProvider` and `IntlProvider` ever since it was added to the starter. So this will affect no project.

> `CurrentUserProvider` must be beneath or parallel to `ErrorDialogHandler`

This on the other hand affects all projects. But I think, as @fraxachun said, it's better to show "Cannot load user" instead nothing